morphia/src/main/java/org/mongodb/morphia/query/FieldEndImpl.java

```diff
@@ -142,7 +142,13 @@ public T doesNotHaveThisElement(final Object val) {
     @Override
     public T hasThisElement(final Object val) {
```

Comment:
Could 'val' be a DBObject with the query criteria?  Issue 970 implies that it could not.